src/org/zaproxy/zap/extension/ascanrules/CommandInjectionPlugin.java

```diff
@@ -623,5 +660,34 @@ private static double getResponseTimeAverage(List<Long> responseTimes) {
 
         return result / responseTimes.size();
     }
+	
+    /*Generate payload variants for uninitialized variable waf bypass
+     *https://www.secjuice.com/web-application-firewall-waf-evasion/
+     *
+     * @param cmd the cmd to insert uninitialized variable
+     */
+    private static String insertUninitVar(String cmd){
+        //get a random 1-10 lowercase letters long variable name
+        int randLength = ThreadLocalRandom.current().nextInt(1,11);
+        byte[] array = new byte[randLength+1];
+        //$xxxxxx
+        array[0]='$';
+        for(int i=1;i<randLength+1;++i){
+            array[i]=(byte)ThreadLocalRandom.current().nextInt(97,123);
+        }
+        String generatedName = new String(array);
+        StringBuffer res = new StringBuffer(cmd);
+
+        //insert variable before each space and '/' in the path
+        for(int i=0;i<res.length();++i){
+            if(res.charAt(i)==' '||res.charAt(i)=='/'){
+                res.insert(i,generatedName);
+                i+=randLength+1;
+            }
+        }

```

Comment:
This loop could be replaced with a single line using chained `replaceAll`, something like:
```
	public static void main(String[] args) {
		String ins = "\\$u";
		String cmd = "cat /etc/passwd";

		System.out.println(cmd.replaceAll("\\s", ins+" ").replaceAll("\\/", "\\/"+ins));
	}
```

`cat$u /$uetc/$upasswd`